idseq_pipeline/commands/non_host_alignment_functions.py

```diff
@@ -883,30 +926,40 @@ def call_hit_level(hits):
                 format(count, input_m8))
     write_to_log("Summarized hits for all {} read ids from {}.".format(
         count, input_m8))
+
+    # Generate output files
     emitted = set()
     with open(output_m8, "wb") as outf:
         with open(output_summary, "wb") as outf_sum:
-            for read_id, accession_id, _percent_id, _alignment_length, e_value, line in iterate_m8(
-                    input_m8, "call_hits_m8_emit_deduped_and_summarized_hits"):
-                if read_id not in emitted:
-                    best_e_value, (hit_level, taxid,
-                                   best_accession_id) = summary[read_id]
-                    if best_e_value == e_value and best_accession_id in (
-                            None, accession_id):
-                        emitted.add(read_id)
-                        outf.write(line)
-                        outf_sum.write(
-                            "{read_id}\t{hit_level}\t{taxid}\n".format(
-                                read_id=read_id,
-                                hit_level=hit_level,
-                                taxid=taxid))
+            # Iterator over the lines of the m8 file
```

Comment:
"[...] Emit only hits that had the best e-value. TODO: Emit all hits within a fixed margin of the best e-value. This change may need to be accompanied by a change to GSNAP/RAPSearch parameters."

Reply:
addressed